### PR TITLE
HubSpot Client

### PR DESCRIPTION
**Pull request by manas-vessel**
## Description

_Detailed description of the changes and the intent behind them_

## Checklist

- [ ] Changes are covered by tests if behavior has been changed or added
- [ ] Tests have 100% coverage
- [ ] If code changes were made, the version in `package.json` has been bumped (matching semver)

## Resolves

If the PR resolves an open issue tag it here. For example: `Resolves #34`

### CODE REVIEW

File: src/platforms/hubspot/schemas.ts [hunk 1]
```diff
@@ -258,6 +247,7 @@ const notePropertiesSchema = z
   .object({
     hubspot_owner_id: hubspotIdSchema,
     hs_note_body: z.string(),
+    hs_timestamp: z.string(),
```
Comment (zkirby) [inline src/platforms/hubspot/schemas.ts:250]: just making sure, this is required? For some reason I thought hubspot would set it for you if not

File: src/sdk/client.ts [hunk 2]
```diff
@@ -67,28 +67,22 @@ export const makeRequestFactory = (
...
 
+      if (!response.ok) {
         throw new IntegrationError('HTTP error in client', {
           type: 'http',
-          body:
-            guard(
-              () => JSON.parse(text),
-              (err) => err instanceof SyntaxError,
-            ) ?? text,
+          body,
```
Comment (zkirby) [inline src/sdk/client.ts:79]: just want to make sure you know, this is now `body: { body: text }` in the case that it's a text response. Not sure if that's intentional

**Review by zkirby — APPROVED** [review]

File: src/platforms/hubspot/index.ts [hunk 3]
```diff
@@ -11,7 +12,10 @@ export default platform('hubspot', {
     tokenUrl: `https://api.hubapi.com/oauth/v1/token`,
     tokenAuth: 'body',
     isRetryable: async ({ response }) => {
-      return (await response.json()).category === 'EXPIRED_AUTHENTICATION';
+      if (response.status === 204) return false;
+      const checkExpiredAuth = async () =>
+        (await response.json()).category === 'EXPIRED_AUTHENTICATION';
```
[the remark this answers is not in the document]
Comment (manas-vessel) [reply src/platforms/hubspot/index.ts:17]: It's defaulting to false below, so it won't retry if the response isn't json (which means probably some non-auth related error)

File: src/platforms/hubspot/schemas.ts (hunk 1 above)
Comment (manas-vessel) [reply src/platforms/hubspot/schemas.ts:250]: I have it `.partial()` below, not required in the updates/creates

File: src/sdk/client.ts (hunk 2 above)
Comment (manas-vessel) [reply src/sdk/client.ts:79]: Yeah, I wasn't sure how we were using this — @zlalvani, is it safe to have it this way?

**Pull request merged**